be1-go/hub/organizer/mod.go

```diff
@@ -291,6 +295,30 @@ func (h *Hub) handleMessageFromClient(incomingMessage *socket.IncomingMessage) e
 	return nil
 }
 
+func verifySignature(publicKey64 string, signature64 string, data64 string) error {
+	pk, err := base64.URLEncoding.DecodeString(publicKey64)
+	if err != nil {
+		return xerrors.Errorf("failed to decode the public key")
+	}
+	var point = suite.Point()
+	err = point.UnmarshalBinary(pk)
+	if err != nil {
+		return xerrors.Errorf("failed to unmarshal the public key: %v", err)
+	}
+
+	signature, err := base64.URLEncoding.DecodeString(signature64)
+	if err != nil {
+		return xerrors.Errorf("failed to decode the signature: %v", err)
+	}
+
+	data, err := base64.URLEncoding.DecodeString(data64)
+	if err != nil {
+		return xerrors.Errorf("failed to decode the data: %v", err)
+	}
+
+	return schnorr.VerifyWithChecks(crypto.Suite, pk, data, signature)
```

Comment:
I'm not a big fan of returning an invocation of a function. It makes the code harder to follow and can make debugging harder, with less meaningful error message.